pkg/controller/dataimportcron-controller.go

```diff
@@ -87,7 +97,7 @@ func (r *DataImportCronReconciler) Reconcile(ctx context.Context, req reconcile.
 	dataImportCron := &cdiv1.DataImportCron{}
 	if err := r.client.Get(ctx, req.NamespacedName, dataImportCron); err != nil {
 		if k8serrors.IsNotFound(err) {
-			return reconcile.Result{}, nil
+			return reconcile.Result{}, r.checkFailingDataImportCrons()
```

Comment:
why do we care about it if DataImportCron not found?

Reply:
If we hit some err in cleanup() and thus requeue the reconcile request, for example
(Otherwise metric value not updated to reflect cleanup)

Reply:
Ok, so why not do a single call to checkFailingDataImportCrons() in the Reconcile() instead of calling it all over?

Reply:
We will call it once per Reconcile() as it is right now if I'm not mistaken:
- IsNotFound case backs off when done (cleanup() wont be called)
- cleanup also backs off when done (and only executed if wasn't IsNotFound)
- update() is only called if we never reached the two above (cleanup, IsNotFound)

Reply:
I meant that it's better that the single call to checkFailingDataImportCrons() will appear only in Reconcile().

Reply:
We can't do a single call because there are a few code branches that need to call it (the ones listed above)

Reply:
Done, thanks for the tip, defer used in cleanup()